### PR TITLE
What's new - OpenSea security provider

### LINKED ISSUE #16636: What's New for the transaction security provider
## Background
We want to announce the new security provider feature to users. We should follow a similar design that was used for the Smart Transactions (screenshot).

## Acceptance Criterias
1. Clicking on `Enable security provider` should turn the feature on. 
2. There should have an `X` on top as the other What's new modals have so users are able to close it.

## Reference
[Design on figma](https://www.figma.com/file/Q69vHaA7Or7Rxz4vdzjVHC/Support-for-tx-security-providers-(MVP)?node-id=4806%3A27316&t=r03VOmHXdcPZlRBn-0#322791664)

**Final Design:** 
Image

### PR DESCRIPTION
**Pull request by filipsekulic**
## Explanation
Created a `What's new` popover for the `OpenSea security provider`.

* Fixes #16636

## Screenshots/Screencaps

https://user-images.githubusercontent.com/92310504/212569864-855e1d12-abbb-407d-82d3-f46117a2d216.mov

## Steps to reproduce
Before starting the app, in the `.metamaskrc` file set the feature flag to 1 - `TRANSACTION_SECURITY_PROVIDER=1`.

1. Create a new wallet or import an existing wallet
2. Close the portfolio popup and refresh the page
3. The popup will appear
4. Clicking on `Enable security provider` will enable the `OpenSea provider`, which can be seen in the `Experimental` tab inside settings, and the popup will no longer be shown to a user
5. The popup will no longer be shown to a user if clicked on:
   - `Not now` button
   -  `X` (close button)
   -  outside the popover (wherever on the home screen)

### CODE REVIEW

**Comment by dzfjz** [comment]
Verified by QA

File: ui/pages/home/home.component.js [hunk 1]
```diff
@@ -550,6 +554,94 @@ export default class Home extends PureComponent {
...
+            color={COLORS.TEXT_ALTERNATIVE}
+            marginTop={4}
+          >
+            {t('alwaysBeSureTo')}
+          </Typography>
+        </Box>
+      </Popover>
+    );
+  }
+
```
Comment (jpuri) [inline ui/pages/home/home.component.js:644]: Home component is hugs, lets extract this into separate component.

File: ui/pages/home/home.component.js [hunk 2]
```diff
@@ -598,6 +691,11 @@ export default class Home extends PureComponent {
         />
         <div className="home__container">
           {showWhatsNew ? <WhatsNewPopup onClose={hideWhatsNewPopup} /> : null}
+          {process.env.TRANSACTION_SECURITY_PROVIDER &&
+          showWhatsNew &&
+          !openSeaTransactionSecurityProviderPopoverHasBeenShown
+            ? this.renderOpenSeaProviderPopover()
+            : null}
```
Comment (jpuri) [inline ui/pages/home/home.component.js:698]: Conditions above can be included into new component to render security provider. ref: https://github.com/MetaMask/metamask-extension/pull/16831/files#r1072384314

File: ui/pages/home/home.component.js (hunk 1 above)
Comment (filipsekulic) [reply ui/pages/home/home.component.js]: Fixed here: b693772adac938c037cc398851c79bc16a7c0610

File: ui/pages/home/home.component.js (hunk 2 above)
Comment (filipsekulic) [reply ui/pages/home/home.component.js]: Fixed here: b693772adac938c037cc398851c79bc16a7c0610

**Comment by seaona** [comment]
@filipsekulic  I've noticed that the OpenSea prompt is displayed behind the Portfolio Notification. Should the OpenSea prevail over the portfolio prompt?

https://user-images.githubusercontent.com/54408225/213771268-6d89a726-cef9-4ea7-87b3-1709c8e82db1.mp4

**Review by jpuri — DISMISSED** [review]

**Comment by filipsekulic** [comment]
> @filipsekulic I've noticed that the OpenSea prompt is displayed behind the Portfolio Notification. Should the OpenSea prevail over the portfolio prompt?
> 
>  opensea-portfolio-prompts.mp4

Hey @seaona!
I did not notice the issue you mentioned. Did you pull the latest changes?

https://user-images.githubusercontent.com/92310504/213995828-edf2cc43-8e19-436f-8df6-4f4569d99afe.mov

**Comment by bschorchit** [comment]
I think this happen when you don't close the Portfolio notification (I've seen it happening in other modals as well). This might be being addressed already here https://github.com/MetaMask/metamask-extension/issues/17398

**Comment by bschorchit** [comment]
A last minute request (sorry!), @filipsekulic could we update this so the user is directed to this section in settings (screenshot) after he/she clicks the `Enable security provider` button?

That will help both give a positive feedback to the user that the feature has been enable and also it'll give visibility to the user to the more detailed explanation and privacy implications of the feature that is within settings. 


Screenshot 2023-02-14 at 22 51 50

**Comment by filipsekulic** [comment]
> A last minute request (sorry!), @filipsekulic could we update this so the user is directed to this section in settings (screenshot) after he/she clicks the `Enable security provider` button?
> 
> That will help both give a positive feedback to the user that the feature has been enable and also it'll give visibility to the user to the more detailed explanation and privacy implications of the feature that is within settings.
> 
> Screenshot 2023-02-14 at 22 51 50

Hey @bschorchit!
No problem. I fixed it. :blush: 
You can check it here:

https://user-images.githubusercontent.com/92310504/219605421-27e3d449-6c70-4e5e-8fab-07c9888dccdf.mov

File: ui/components/app/open-sea-whats-new-popover/open-sea-whats-new-popover.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            color={TextColor.textAlternative}
+            marginTop={4}
+          >
+            {t('alwaysBeSureTo')}
+          </Text>
+        </Box>
+      </Popover>
+    )
+  );
+}
```
Comment (jpuri) [inline ui/components/app/open-sea-whats-new-popover/open-sea-whats-new-popover.js:124]: Will be nice to have some test coverage for this component.

File: ui/store/actions.ts [hunk 4]
```diff
@@ -4515,6 +4516,20 @@ export function setFirstTimeUsedNetwork(chainId: string) {
...
+  void,
+  MetaMaskReduxState,
+  unknown,
+  AnyAction
+> {
+  return async (dispatch) => {
+    await submitRequestToBackground(
+      'setOpenSeaTransactionSecurityProviderPopoverHasBeenShown',
+    );
+    await forceUpdateMetamaskState(dispatch);
```
Comment (jpuri) [inline ui/store/actions.ts:4529]: Can you plz detail why is this required `await forceUpdateMetamaskState(dispatch);` ?

**Review by jpuri — DISMISSED** [review]

File: ui/components/app/open-sea-whats-new-popover/open-sea-whats-new-popover.js (hunk 3 above)
Comment (filipsekulic) [reply ui/components/app/open-sea-whats-new-popover/open-sea-whats-new-popover.js:124]: I agree. I created this PR some time ago when it wasn't mandatory... 
I'll cover it with tests after I finish all the tickets regarding the security provider feature. For now, I think we can push forward this PR and not block it, if you agree.

File: ui/store/actions.ts (hunk 4 above)
Comment (filipsekulic) [reply ui/store/actions.ts]: Good question and good catch! I removed it now and everything works fine. Not sure why I added it before, like something wasn't working properly and I was looking into other actions that were implemented in the same way...
It is fixed now: 85964c7c518d262946fc6d109f44c9d4863e23b0

**Review by jpuri — APPROVED** [review]

File: app/scripts/controllers/preferences.js [hunk 5]
```diff
@@ -70,6 +70,7 @@ export default class PreferencesController {
         : LedgerTransportTypes.u2f,
       transactionSecurityCheckEnabled: false,
       theme: ThemeType.os,
+      openSeaTransactionSecurityProviderPopoverHasBeenShown: false,
```
Comment (digiwand) [inline app/scripts/controllers/preferences.js:73]: [suggestion] for boolean names there are suggested prefixes:  "is", "has", "should", "can", "did", "will" (pulled from ESLint rule we are not using). It can help with readability by understanding the intent of the return value reading left to right.

we could update `openSeaTransactionSecurityProviderPopoverHasBeenShown` -> `hasShownOpenSeaTransactionSecurityProviderPopover`

ref ESLint rule: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md#enforce-that-boolean-variables-are-prefixed-with-an-allowed-verb

**Review by digiwand — APPROVED** [review]
Nice!  I left a minor improvement suggestion

File: ui/components/app/open-sea-whats-new-popover/open-sea-whats-new-popover.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        className="smart-transactions-popover"
+        onClose={() =>
+          dispatch(setOpenSeaTransactionSecurityProviderPopoverHasBeenShown())
+        }
+      >
+        <Box
+          paddingRight={6}
+          paddingLeft={6}
+          paddingTop={0}
+          paddingBottom={0}
```
Comment (digiwand) [inline ui/components/app/open-sea-whats-new-popover/open-sea-whats-new-popover.js:88]:
```suggestion
```

would removing these two lines produce the same result?
